RickAndMortyFunApp/NetworkServices/NetworkService.swift

```diff
@@ -6,56 +6,97 @@
 //  Copyright © 2019 Rusłan Chamski. All rights reserved.
 //
 
+import Alamofire
 import Foundation
 
 enum NetworkError: Error {
     case invalidURL
 }
 
+enum NetworkRouter: URLRequestConvertible {
+    static let baseURLString = "https://rickandmortyapi.com/api/character/"
+
+    case get(Int)
+    case create([String: Any])
+    case delete(Int)
+
+    func asURLRequest() throws -> URLRequest {
+        var method: HTTPMethod {
+            switch self {
+            case .get:
+                return .get
+            case .create:
+                return .post
+            case .delete:
+                return .delete
+            }
+        }
+
+        let params: ([String: Any]?) = {
+            switch self {
+            case .get, .delete:
+                return nil
+            case .create(let newTodo):
+                return (newTodo)
+            }
+        }()
+
+        let url: URL = {
+            var components = URLComponents(string: NetworkRouter.baseURLString)
+            switch self {
+            case .get(let page):
+                components?.queryItems = [
+                    URLQueryItem(name: "page", value: String(page))
```

Comment:
When we use Alamofire this way, then we don't need add parameters as URL components. What I mean is that parameters should look like:
```
let params: ([String: Any]?) = {
     switch self {
     case .getCharacters(let page):
         return ["page": String(page)]
     }
}()
```
So later, when we create url request, we choose to use URL encoding for parameters. That's how it sets instead of URLQueryItem.

Then url:
```
let url: URL = {
    let relativePath: String
    switch self {
    case .getCharacters:
        relativePath = "character"
    }
    let url = baseURL.appendingPathComponent(relativePath)
    return url
}()
```
That's how we set character as a part of url